src/viam/resource/easy_resource.py

```diff
@@ -25,6 +28,50 @@ def _parse_model(orig: Union[str, Model]) -> Model:
     return Model(ModelFamily(*family), name)
 
 
+def _create_stub_fn(name: str, is_async: bool) -> Callable:
+    """
+    This creates a sync or async stub function which returns a MethodNotImplementedError.
+    The stub_model decorator uses these to stub out an abstract base class.
+    """
+    if is_async:
+        # note: this is a pyright bug https://github.com/microsoft/pyright/issues/2136
+        async def stub_fn(*args, **kwargs):  # pyright: ignore [reportRedeclaration]
+            logger.info(f'{name} not implemented')
+            raise MethodNotImplementedError(name)
+    else:
+        def stub_fn(*args, **kwargs):
+            logger.info(f'{name} not implemented')
+            raise MethodNotImplementedError(name)
+
+    stub_fn.__name__ = f'{name}_stub'
+    return stub_fn
+
+
+def stub_model(cls: ABCMeta) -> ABCMeta:
+    """
+    Class decorator which adds error implementations of abstract functions. This means they will fail
+    when called, rather than the default where they fail when instantiated. This is intended for developers
+    who want to build and test incrementally, not for production use.
+
+    Example:
+
+        @stub_model
+        class MyMotor(Motor, EasyResource):
+            MODEL = 'viam:motor:easy-resource-example'
+
+    Normally this class would fail to instantiate. With the decorator, it will succeed but the unimplemented
+    methods will throw errors at runtime.
+    """
+    for attr in cls.__abstractmethods__:
+        val = getattr(cls, attr)
+        is_async = inspect.iscoroutinefunction(val)
+        stub_fn = _create_stub_fn(attr, is_async)
+        setattr(cls, attr, stub_fn)
+        logger.debug('patched %s.%s with %s', cls, attr, stub_fn)
+        cls.__abstractmethods__ -= {attr}
```

Comment:
Updating the object while you're iterating over it... dangerous :P

Reply:
:grimacing: eek solved in 6b3a1a50